sceptre/plan/actions.py

```diff
@@ -1000,3 +1000,144 @@ def diff(self, stack_differ):
         :rtype: sceptre.diffing.stack_differ.StackDiff
         """
         return stack_differ.diff(self)
+
+    @add_stack_hooks
+    def drift_detect(self) -> Dict[str, str]:
+        """
+        Show stack drift for a running stack.
+
+        :returns: The stack drift detection status.
+        If the stack does not exist, we return a detection and
+        stack drift status of STACK_DOES_NOT_EXIST.
+        If drift detection times out after 5 minutes, we return
+        TIMED_OUT.
+        """
+        try:
+            self._get_status()
+        except StackDoesNotExistError:
+            self.logger.info(f"{self.stack.name} - Does not exist.")
+            return {
+                "DetectionStatus": "STACK_DOES_NOT_EXIST",
+                "StackDriftStatus": "STACK_DOES_NOT_EXIST"
+            }
+
+        response = self._detect_stack_drift()
+        detection_id = response["StackDriftDetectionId"]
+
+        try:
+            response = self._wait_for_drift_status(detection_id)
+        except TimeoutError as exc:
+            self.logger.info(f"{self.stack.name} - {exc}")
+            response = {
+                "DetectionStatus": "TIMED_OUT",
+                "StackDriftStatus": "TIMED_OUT"
+            }
+
+        return response
+
+    @add_stack_hooks
+    def drift_show(self) -> Tuple[str, dict]:
+        """
+        Detect drift status on stacks.
+
+        :returns: The detection status and resource drifts.
+        """
+        response = self.drift_detect()
+        detection_status = response["DetectionStatus"]
+
+        if detection_status in ["DETECTION_COMPLETE", "DETECTION_FAILED"]:
+            response = self._describe_stack_resource_drifts()
+        elif detection_status in ["TIMED_OUT", "STACK_DOES_NOT_EXIST"]:
+            response = {
+                "StackResourceDriftStatus": detection_status
+            }
+        else:
+            raise Exception("Not expected to be reachable")
+
+        return (detection_status, response)
+
+    def _wait_for_drift_status(self, detection_id: str) -> dict:
+        """
+        Waits for drift detection to complete.
+
+        :param detection_id: The drift detection ID.
+        :returns: The response from describe_stack_drift_detection_status.
+        """
+        timeout = 300
```

Comment:
magical number.  at the very least we should define a descriptive variable for this default value.  would be even nicer if users could override this value from command line.

Reply:
I think that this timeout is supposed to be unreachable and if any user actually does reach it in any scenario other than an AWS outage then a new value should be hard coded. Therefore, I can't personally see the value in a configuration option for it.

Reply:
ok, your argument sounds reasonable.